File: fundamentos-big-data/spark/ejemplos/learning-spark-2nd-edition/capitulo2/el-quijote/python/src/elquijote.py
# -*- coding: utf-8 -*-
# Importamos las librerias necesarias
import sys
from pyspark.sql import SparkSession

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Uso: elquijote <archivo>", file=sys.stderr)
        sys.exit(-1)
    # Construimos el SparkSession, si no existe se crea una instancia
    spark = (SparkSession
        .builder
        .appName("PythonMnMCount")
        .getOrCreate())
    
    # Obtenemos el nombre del fichero de datos de los argumentos de la linea de comandos
    quijote_file = sys.argv[1]

    # Leemos el fichero y persistimos el DF que generamos
    quijote_df = (spark.read.text(quijote_file))
    quijote_df.persist()

    # Mostrar numero de filas
    quijote_df.count()

    # Mostrar 20 primeras filas
    quijote_df.show()

    # Mostrar 5 primeras filas
    quijote_df.show(5)

    # Mostrar 5 primeras filas sin truncar
    quijote_df.show(5, False)

    # Mostar filas sin truncar y en vertical
    quijote_df.show(truncate=False, vertical=True)

    # Mostrar 2 primeras filas truncadas y en vertical
    quijote_df.show(2, vertical=True)

    # Obtener primera fila del DF
    quijote_df.head()
    # Equivalente
    quijote_df.first()

    # Obtener 5 primeras filas del DF
    quijote_df.head(5)
    # Equivalente
    quijote_df.take(5)

    # Detenemos el SparkSession
    spark.stop()

File: fundamentos-big-data/spark/ejemplos/Scala/learning-spark-2nd-edition/capitulo2/el-quijote/src/main/scala/capitulo2A/elquijote.scala
package main.scala.capitulo2A

import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.functions._

object QuijoteCount {
    def main(args: Array[String]) {
        //Comprobamos si el numero de argumentos es correcto
        if(args.length < 1) {
            print("Uso: QuijoteCount <archivo_el_quijote.txt>")
            sys.exit(1)
        }
        //Construimos el SparkSession, si no existe se crea una instancia
        val spark = SparkSession
            .builder
            .appName("QuijoteCount")
            .getOrCreate()
        
        // Obtenemos el nombre del fichero de datos de los argumentos de la linea de comandos
        val quijoteFile = args(0)

        // Leemos el fichero txt, lo convertimos a un Dataset y lo persistimos
        val quijoteDs = spark.read.text(quijoteFile)
        quijoteDs.persist()
        
        // Mostrar numero de filas
        quijoteDs.count()

        // Mostrar 20 primeras filas
        quijoteDs.show()
    
        // Mostrar 5 primeras filas
        quijoteDs.show(5)
    
        // Mostrar 5 primeras filas sin truncar
        quijoteDs.show(5, false)
    
        // Mostar filas sin truncar
        quijoteDs.show(false)
    
        // Obtener primera fila del DF
        quijoteDs.head()
        // Equivalente
        quijoteDs.first()
    
        // Obtener 5 primeras filas del DF
        quijoteDs.head(5)
        // Equivalente
        quijoteDs.take(5)
            
        // Detenemos el SparkSession
        spark.stop()
    }
}